tcs/csp_solver_mspt_receiver.cpp

```diff
@@ -619,422 +650,169 @@ void C_mspt_receiver::call(const C_csp_weatherreader::S_outputs &weather,
 		m_dot_htf_max = fmin(m_m_dot_htf_max, m_dot_iscc_max);
 	}
     
-	double q_abs_sum = 0.0;
-	double err_od = 999.0;	// Reset error before iteration
-
-	// 15 continue -> TRNSYS command - replace
+	
 	if (field_eff < m_eta_field_iter_prev && m_od_control < 1.0)
 	{	// Suggests controller applied defocus, so reset *controller* defocus
 		m_od_control = fmin(m_od_control + (1.0 - field_eff / m_eta_field_iter_prev), 1.0);
 	}
 
-	do
-	{
-		if( rec_is_off )
-			break;
 
-		field_eff_adj = field_eff*m_od_control;
+	s_steady_state_soln soln, soln_actual, soln_clearsky;
```

Comment:
You don't need to change this (as it is a bit pedantic), so just for future consideration:
Per [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Type_Names), change to:
		SteadyStateSolution steady_state_solution, etc.
		<or>
		SteadyStateSolution solution, solution_actual, solution_clearsky;

The variable names are longer and less compact, but clarity over conciseness.

Reply:
note: it took out the "or" between the two 'SteadyStateSolution' lines. Not sure why.